src/java.base/share/classes/java/math/MutableBigInteger.java

```diff
@@ -109,8 +110,28 @@ class MutableBigInteger {
      * the int val.
      */
     MutableBigInteger(int val) {
+        init(val);
+    }
+
+    /**
+     * Construct a new MutableBigInteger with a magnitude specified by
+     * the long val.
+     */
+    MutableBigInteger(long val) {
+        int hi = (int) (val >>> 32);
+        if (hi == 0) {
+            init((int) val);
+        } else {
+            value = new int[2];
+            intLen = 2;
+            value[0] = hi;
+            value[1] = (int) val;
```

Comment:
```suggestion
            value = new int[] {hi, (int) val}
            intLen = 2;
```

Reply:
In this case, should the same method be applied in the inizialization with an int?

Reply:
You mean, in `init()`? Then yes, please.